overlord/hookstate/hookmgr.go

```diff
@@ -343,8 +344,9 @@ func (m *HookManager) runHookGuardForRestarting(context *Context) error {
 }
 
 func (m *HookManager) runHook(context *Context, snapst *snapstate.SnapState, hooksup *HookSetup, tomb *tomb.Tomb) error {
-	mustHijack := m.hijacked(hooksup.Hook, hooksup.Snap) != nil
+	mustHijack := context.IsSnapHook() && m.hijacked(hooksup.Hook, hooksup.Snap) != nil
```

Comment:
Are we sure that the "hijack" scenarion would not affect component hooks? Say, while installing in seeding stage. cc @pedronis

Reply:
although it's a general mechanism we use Hijack only for configure for core, so I think for now it's ok to ignore components as we would likely need it only we added components to snapd snap itself

Reply:
I added a comment highlighting this.